citar-file.el

```diff
@@ -124,46 +129,118 @@ Example: ':/path/to/test.pdf:PDF'."
          (mapcar (apply-partially #'expand-file-name fn) dirs)))
      parts)))
 
-(defun citar-file--possible-names (key dirs extensions &optional entry find-additional)
-  "Possible names for files correponding to KEY, ENTRY with EXTENSIONS in DIRS."
-    (let* ((filematch (when find-additional
-                        (format "\\`%s\\(?:%s.*\\)?\\.\\(?:%s\\)\\'"
-                                (regexp-quote key)
-                                (if (eq t find-additional) "" find-additional)
-                                (mapconcat #'regexp-quote extensions "\\|"))))
-           (results-key (seq-mapcat
-                         (lambda (dir)
-                           (if filematch
-                               (directory-files dir t filematch)
-                             (seq-map (lambda (ext)
-                                        (expand-file-name (concat key "." ext) dir))
-                                      extensions)))
-                         dirs))
-           (file-field (citar-get-value
-                        citar-file-variable entry))
-           (results-file
-            (when file-field
-              (seq-mapcat
-               (lambda (func)
-                 (funcall
-                  func
-                  ;; Make sure this arg is non-nil.
-                  (or dirs "/")
-                  file-field))
-               citar-file-parser-functions))))
-      (delete-dups (append results-key results-file))))
+(defun citar-file--parse-file-field (entry fieldname &optional dirs)
+  "Return files listed in FIELDNAME of ENTRY.
+File names are expanded relative to the elements of DIRS."
+  (unless dirs (setq dirs (list "/")))  ; make sure DIRS is non-nil
+  (when-let ((filefield (citar-get-value fieldname entry)))
+    (seq-mapcat (lambda (parser)
+                  (funcall parser dirs filefield))
+                citar-file-parser-functions)))
+
+(defun citar-file--filename-regexp (keys extensions &optional find-additional)
```

Comment:
So this is a function which _creates_ a regexp? Perhaps the below would be more clear?

```suggestion
(defun citar-file--filename-make-regexp (keys extensions &optional find-additional)
```